Use experimental Miso components support

georgefst opened the pull request:
See https://github.com/dmjio/miso/pull/766.

This works, until the last commit, where we try to actually add a component, which causes the app to crash with `TypeError: obj.mount is not a function`. The Miso components branch does not yet support Wasm (or `jsaddle-warp`) due to a need for synchronous callbacks, but this is being worked on at this very moment.

If upstream support does appear in the next week or so, then I propose that we use this as the basis for the next stage of frontend development, i.e. we rebase [my WIP actions and eval work](https://github.com/hackworthltd/primer/compare/main...georgefst/frontend-eval) on top of this. Both the actions and eval panels would benefit from being components since they have a natural notion of internal state. I would otherwise have considered fake components as outlined [here](https://github.com/FPtje/miso-component-example), but I'd say we're better off embracing the future. The success of the frontend rewrite so far has been built on using bleeding edge features and working closely with upstream maintainers. I see no reason to stop now!

georgefst committed:
chore: Use `MisoString` instead of `Text` where possible

Miso currently uses `type MisoString = Text` when targeting Wasm, but will soon change to `JSString` for better performance, as is already the case when targeting JavaScript. So this is future-proofing.

Signed-off-by: George Thomas <georgefsthomas@gmail.com>

dhess commented:
We're in no hurry, so we might as well wait a bit for the components Wasm support.

georgefst commented:
Well, we didn't have to wait very long! Miso's components now work with `jsaddle-warp` and `jsaddle-wasm`. Both use async callbacks for mounting which is unsafe in theory. In the Wasm case, sync callbacks are [coming soon](https://gitlab.haskell.org/haskell-wasm/ghc-wasm-meta/-/tree/wip/wasm-jsffi-sync-export) but I think there's some work needed in `jsaddle` and `miso` for it to filter through.

georgefst committed:
chore: Use Miso with components support

Other unreleased changes include better Wasm support.

This does not, or at least should not, yet cause any changes in functionality.

Signed-off-by: George Thomas <georgefsthomas@gmail.com>

georgefst committed:
add a dummy Miso component

Signed-off-by: George Thomas <georgefsthomas@gmail.com>

dhess commented:
We've already switched to Miso components.
